src/qililab/automatic_calibration/calibration_node.py

```diff
@@ -0,0 +1,690 @@
+                self.nb_path,
+            )
+            raise IncorrectCalibrationOutput(f"No output found, check automatic-calibration notebook in {self.nb_path}")
+
+        if len(outputs_lines) > 1:
+            logger.error(
+                "Aborting execution. More than one output found, please output the results once in %s",
+                self.nb_path,
+            )
+            raise IncorrectCalibrationOutput(f"More than one output found in {self.nb_path}")
+
+        # When only one line of outputs, use that one:
+        return self._from_logger_string_to_output_dict(outputs_lines[0], self.nb_path)
+
+    def _from_logger_string_to_output_dict(self, logger_string, input_path):
+        """Returns the output dictionary from the logger string, or raises errors if ``logger_string`` doesn't follow the expected format.
+
+        Args
+            logger_string (str): The logger string containing the output dictionary to extract.
+            input_path (str): Path of the notebook that generated the output, to raise errors.
+
+        Returns:
+            dict: Kwargs for the output parameters of the notebook.
+
+        Raises:
+            IncorrectCalibrationOutput: In case no outputs, incorrect outputs or multiple outputs where found. Incorrect outputs are those that do not contain `check_parameters` or is empty.
+        """
+        logger_splitted = logger_string.split(logger_output_start)
+
+        # In case something unexpected happened with the output we raise an error
+        if len(logger_splitted) < 2:
+            logger.error(
+                "Aborting execution. No output found, check the automatic-calibration output cell is implemented in %s",
+                input_path,
+            )
+            raise IncorrectCalibrationOutput(f"No output found, check automatic-calibration notebook in {input_path}")
+        if len(logger_splitted) > 2:
+            logger.error(
+                "Aborting execution. More than one output found, please output the results once in %s",
+                input_path,
+            )
+            raise IncorrectCalibrationOutput(f"More than one output found in {input_path}")
+
+        # This next line is for taking into account other encodings, where special characters get `\\` in front.
+        clean_data = logger_splitted[1].split("\\n")[0].replace('\\"', '"')
+
+        logger_outputs_string = clean_data.split("\n")[0]
+        out_dict = json.loads(logger_outputs_string)
+
+        if "check_parameters" not in out_dict or out_dict["check_parameters"] == {}:
+            logger.error(
+                "Aborting execution. No 'check_parameters' dictionary or its empty in the output cell implemented in %s",
+                input_path,
+            )
+            raise IncorrectCalibrationOutput(
+                f"Empty output found in {input_path}, output must have key and value 'check_parameters'."
+            )
+        return out_dict
+
```

Comment:
Use `os.path.join` when dealing with directories.


```suggestion
        path = os.path.join(self.nb_folder, self.node_id)
```

Reply:
We'll take a look into this!

Reply:
@IsaacLA12

Reply:
Commited suggestion, thanks!

Reply:
Actually in various places!